cmake/modules/extensions.cmake

```
@@ -1559,6 +1559,66 @@ function(zephyr_syscall_header_ifdef feature_toggle)
   endif()
 endfunction()
 
+# Verify blobs fetched using west. If the sha256 checksum isn't valid, a warning/
+# fatal error message is printed (depends on REQUIRED flag).
+# Paths are relative to the current module's zephyr/blobs directory.
+#
+# Usage:
+#   zephyr_blobs_verify([PATHS path [paths...]]
+#                       [REQUIRED])
+#
+# Example:
+# zephyr_blobs_verify(REQUIRED)           # verify all blobs and fail on error
+# zephyr_blobs_verify(PATHS img/file.bin) # verify a single path
+function(zephyr_blobs_verify)
+  if(NOT DEFINED ZEPHYR_CURRENT_MODULE_NAME)
+    message(FATAL_ERROR "Unknown module name when calling ${CMAKE_CURRENT_FUNCTION}")
+  endif()
+
+  if(NOT WEST)
+    return()
+  endif()
+
+  cmake_parse_arguments(BLOBS_VERIFY "REQUIRED" "" "PATHS" ${ARGN})
+
+  execute_process(
+    COMMAND ${WEST} blobs list ${ZEPHYR_CURRENT_MODULE_NAME} --format "{status} {path}"
```

should probably use `abspath` here, so that the exact location is know.
This will also make abspath and relpath support in CMake code cleaner.

```suggestion
    COMMAND ${WEST} blobs list ${ZEPHYR_CURRENT_MODULE_NAME} --format "{status} {abspath}"
```

---

also, including `"{module}"` will provide the module where the missing blob originates, making it possible to inform the user the exact module to use when doing `west blobs fetch <module>`.